packages/enumitem-lite.lua

```diff
@@ -0,0 +1,374 @@
+  -- using \rebox caused an issue sometimes, not sure why, with the bullets
+  -- appearing twice in output... but we can avoid it:
+  -- reboxing an hbox was dumb anyway. We just need to fix its width before
+  -- reinserting it in the text flow.
+  mark.width = SILE.length(stepback)
+  SILE.typesetter:pushHbox(mark)
+  SILE.process(content)
+end
+
+local doNestedList = function (listType, _, content)
+  -- depth
+  local depth = SILE.settings.get("list.current."..listType..".depth") + 1
+
+  -- styling
+  local enumStyle = styles[listType][depth]
+  if not enumStyle then SU.error("List nesting is too deep") end
+
+  -- indent
+  local baseIndent = (depth == 1) and SILE.settings.get("document.parindent").width:absolute() or SILE.measurement("0pt")
+  local listIndent = SILE.settings.get("list."..listType..".leftmargin"):absolute()
+
+  -- processing
+  if not SILE.typesetter:vmode() then
+    SILE.call("par")
+  end
+  SILE.settings.temporarily(function ()
+    SILE.settings.set("list.current."..listType..".depth", depth)
+    SILE.settings.set("current.parindent", SILE.nodefactory.glue())
+    SILE.settings.set("document.parindent", SILE.nodefactory.glue())
+    SILE.settings.set("document.parskip", SILE.settings.get("list.parskip"))
+    local lskip = SILE.settings.get("document.lskip") or SILE.nodefactory.glue()
+    SILE.settings.set("document.lskip", SILE.nodefactory.glue(lskip.width + (baseIndent + listIndent)))
+
+    local counter = 0
+    for i = 1, #content do
+      if type(content[i]) == "table" then
+        if content[i].command == "item" then
+          counter = counter + 1
+          -- Enrich the node with internal properties
+          content[i]._enumitem_ = {
+            style = enumStyle,
+            counter = counter,
+            indent = listIndent,
+          }
+        else
+          enforceListType(content[i].command)
+        end
+        SILE.process({ content[i] })
+        if not SILE.typesetter:vmode() then
+          SILE.call("par")
+        else
+          SILE.typesetter:leaveHmode()
+        end
+      elseif type(content[i]) == "string" then
+        -- All text nodes are ignored in structure tags, but just warn
+        -- if there do not just consist in spaces.
+        local text = trim(content[i])
+        if text ~= "" then SU.warn("Ignored standalone text ("..text..")") end
+      else
+        SU.error("List structure error")
+      end
+    end
+  end)
+
+  if not SILE.typesetter:vmode() then
+      SILE.call("par")
+  else
+    SILE.typesetter:leaveHmode()
+    if not((SILE.settings.get("list.current.itemize.depth")
+        + SILE.settings.get("list.current.enumerate.depth")) > 0)
+    then
+      local g = SILE.settings.get("document.parskip").height - SILE.settings.get("list.parskip").height
+      SILE.typesetter:pushVglue(g)
+    end
+  end
+end
+
+local function init (class, _)
+  class:loadPackage("counters")
+
+  SILE.settings.declare({
+    parameter = "list.current.enumerate.depth",
```

Comment:
That's right, I agree. I usually adhere to the same convention as suggested in my packagess and can't remember why this one eventually came to "list"... I wouldn't mind having these settings changed to e.g. "enumitem.whatever" (and could do the same in my own style-enabled experimental version of that package).

Reply:
What about renaming the package and first component of the settings to something else? I don't know what, `litelists` perhaps. If we don't plan on full feature parity and "spiritual" inheritance from the TeX packages of the same name I don't feel like matching the names really serves us very well. Something that gives us room for other list packages later but also is descriptive enough yet terse to stand on it's own merit...

Reply:
> I don't feel like matching the names really serves us very well.

Er. I don't think (La)TeX has an "enumitem" package. I just called the commands/environments "itemize" and "enumerate" as in LaTeX, because this is what they do, kind of, but beyond that, it's different (the `\item` here takes an argument, and the nesting and skips handling is probably different too - I haven't looked precisely how LaTeX does it, and it's a while since I used it). The package name was coined from the commands. I don't really care, though "lists" might seem vague to me (I'd think to programming structures, like the Penlight `pl.List`, lol). As for the "lite" thing - AFAIC, they are only lite with respect to their customization (e.g. colored bullets, variants for linguistic preferences, etc.) which I covered in my richer implementation, though the generic styling features are (still) in a work-in-progress.

Reply:
> Er. I don't think (La)TeX has an "enumitem" package

Oh lol, it has one by that name actually https://www.ctan.org/pkg/enumitem

I don't even know what this one does (never tried it).

Reply:
Can we at least drop the `-lite` bit? You can add `-full` or `-avdanced` or whatever to your own variant, but I don't think it makes sense to have a variant name like this is core without the bit that is being deviated from.

I suppose `enumitem` is okay, or maybe `lightlists` or `simplelists` (goes with `simpletable`) could work too. If we plan on making any future improvements to this package rather than spawning new ones maybe just calling it `lists` is okay.

Reply:
"enumitem-lite' renamed 'lists"

Reply:
setting names scoped to "lists.xxx" for "lists" package.